Python/PRP/Audio/pyFadeParams.cpp

```diff
@@ -0,0 +1,215 @@
+        return -1;
+    }
+    self->fThis->fLengthInSecs = (float)PyFloat_AsDouble(value);
+    return 0;
+}
+
+static int pyFadeParams_setVolStart(pyFadeParams* self, PyObject* value, void*) {
+    if (!PyFloat_Check(value)) {
+        PyErr_SetString(PyExc_TypeError, "volStart should be a float");
+        return -1;
+    }
+    self->fThis->fVolStart = (float)PyFloat_AsDouble(value);
+    return 0;
+}
+
+static int pyFadeParams_setVolEnd(pyFadeParams* self, PyObject* value, void*) {
+    if (!PyFloat_Check(value)) {
+        PyErr_SetString(PyExc_TypeError, "volEnd should be a float");
+        return -1;
+    }
+    self->fThis->fVolEnd = (float)PyFloat_AsDouble(value);
+    return 0;
+}
+
+static int pyFadeParams_setType(pyFadeParams* self, PyObject* value, void*) {
+    if (!PyInt_Check(value)) {
+        PyErr_SetString(PyExc_TypeError, "type should be an int");
+        return -1;
+    }
+    self->fThis->fType = (unsigned char)PyInt_AsLong(value);
+    return 0;
+}
+
+static int pyFadeParams_setStopWhenDone(pyFadeParams* self, PyObject* value, void*) {
+    if (!PyBool_Check(value)) {
+        PyErr_SetString(PyExc_TypeError, "stopWhenDone should be a boolean");
+        return -1;
+    }
+    self->fThis->fStopWhenDone = PyInt_AsLong(value) != 0;
+    return 0;
+}
+
+static int pyFadeParams_setFadeSoftVol(pyFadeParams* self, PyObject* value, void*) {
+    if (!PyBool_Check(value)) {
+        PyErr_SetString(PyExc_TypeError, "fadeSoftVol should be a boolean");
+        return -1;
+    }
+    self->fThis->fFadeSoftVol = PyInt_AsLong(value) != 0;
+    return 0;
+}
+
+static int pyFadeParams_setCurrTime(pyFadeParams* self, PyObject* value, void*) {
+    if (!PyFloat_Check(value)) {
+        PyErr_SetString(PyExc_TypeError, "currTime should be a float");
+        return -1;
+    }
+    self->fThis->fCurrTime = (float)PyFloat_AsDouble(value);
+    return 0;
+}
+
+static PyGetSetDef pyFadeParams_GetSet[] = {
+    { _pycs("lengthInSecs"), (getter)pyFadeParams_getLengthInSecs,
+      (setter)pyFadeParams_setLengthInSecs, NULL, NULL },
+    { _pycs("volStart"), (getter)pyFadeParams_getVolStart,
+      (setter)pyFadeParams_setVolStart, NULL, NULL },
+    { _pycs("volEnd"), (getter)pyFadeParams_getVolEnd,
+      (setter)pyFadeParams_setVolEnd, NULL, NULL },
+    { _pycs("type"), (getter)pyFadeParams_getType,
+      (setter)pyFadeParams_setType, NULL, NULL },
+    { _pycs("stopWhenDone"), (getter)pyFadeParams_getStopWhenDone,
+      (setter)pyFadeParams_setStopWhenDone, NULL, NULL },
+    { _pycs("fadeSoftVol"), (getter)pyFadeParams_getFadeSoftVol,
+      (setter)pyFadeParams_setFadeSoftVol, NULL, NULL },
+    { _pycs("currTime"), (getter)pyFadeParams_getCurrTime,
+      (setter)pyFadeParams_setCurrTime, NULL, NULL },
+    { NULL, NULL, NULL, NULL, NULL }
+};
+
+PyTypeObject pyFadeParams_Type = {
+    PyVarObject_HEAD_INIT(NULL, 0)
+    "PyHSPlasma.plFadeParams",          /* tp_name */
```

Comment:
Should be `PyHSPlasma.plSound.plFadeParams` based on where the type is instantiated.